pkg/occlient/projects.go

```diff
@@ -0,0 +1,236 @@
+package occlient
+
+import (
+	"fmt"
+	"time"
+
+	projectv1 "github.com/openshift/api/project/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+// GetProject returns project based on the name of the project.Errors related to
+// project not being found or forbidden are translated to nil project for compatibility
+func (c *Client) GetProject(projectName string) (*projectv1.Project, error) {
+	prj, err := c.projectClient.Projects().Get(projectName, metav1.GetOptions{})
+	if err != nil {
+		istatus, ok := err.(kerrors.APIStatus)
+		if ok {
+			status := istatus.Status()
+			if status.Reason == metav1.StatusReasonNotFound || status.Reason == metav1.StatusReasonForbidden {
+				return nil, nil
+			}
+		} else {
+			return nil, err
+		}
+
+	}
+	return prj, err
+
+}
+
+// ListProjects return list of existing projects that user has access to.
+func (c *Client) ListProjects() (*projectv1.ProjectList, error) {
+	return c.projectClient.Projects().List(metav1.ListOptions{})
+}
+
+// GetProjectNames return list of existing project names that user has access to.
+func (c *Client) GetProjectNames() ([]string, error) {
+	projects, err := c.ListProjects()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list projects")
```

Comment:
Similar to what I requested on https://github.com/openshift/odo/pull/4213#discussion_r524912708. Can we not wrap and simply throw the error that `ListProjects` encountered? Also, is there a design pattern we're following here?

Reply:
Fixed